OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -126,4 +126,17 @@ public interface IPreventsAutoTarget
 	{
 		bool PreventsAutoTarget(Actor self, Actor attacker);
 	}
+
+	[Flags]
+	public enum UncloakType
+	{
+		None = 0,
+		Attack = 1,
+		Move = 2,
+		Unload = 4,
+		Infiltrate = 8,
+		Demolish = 16,
+		Damage = 32,
+		Default = Attack | Unload | Infiltrate | Demolish
```

Comment:
This is actually less convenient: if you leave it explicit in the trait the generated docs will list all the types, while 'default' is opaque to modders.